torch/nn/init.py

```diff
@@ -525,3 +527,104 @@ def deprecated_init(*args, **kwargs):
 kaiming_normal = _make_deprecate(kaiming_normal_)
 orthogonal = _make_deprecate(orthogonal_)
 sparse = _make_deprecate(sparse_)
+
+
+v = torch.__version__.split('.')
+# These can also be used as global flags
+_torch_version = (int(v[0]), int(v[1]), int(v[2][0]))
+# Use "init_version" to handle "_init_version". This value corresponds to the last pytorch version that
+# did not have 'init_version' context manager.
+_init_version = (1, 6, 1)
+
+def _parse_version(version: Union[Tuple[int, int, int], str] = None, use_master: bool = False) -> Tuple[int, int, int]:
+    if use_master:
+        return _torch_version
+
+    if version is None:
+        return _init_version
+
+    if isinstance(version, tuple):
+        for x in version:
+            if not isinstance(x, int):
+                raise ValueError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+    elif isinstance(version, str):
+        if version.count('.') != 2:
+            raise ValueError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+        v = version.split('.')
+        v = [v[0], v[1], v[2][0]]
+        for version_type in v:
+            for char in version_type:
+                if not ((str(char) >= '0' and str(char) <= '9') or str(char) == '.'):
+                    raise ValueError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+        v = version.split('.')
+        version = (int(v[0]), int(v[1]), int(v[2][0]))
+    else:
+        raise TypeError("Invalid version, must be a version string (e.g. '1.7.0') or tuple of integers")
+
+    # version, should be less than torch.__version__
+    if version > _torch_version:
+        raise ValueError(f"version {version} should be less than torch version {_torch_version}")
+    return version
+
+
+@contextlib.contextmanager
+def init_version(version=None, use_master=False):
+    r"""Context manager to use a specific version of initialization for `nn.modules`.
+    By default, the pytorch initialization used till version 1.6.1 is used.
+
+    Args:
+        version : Union[Tuple[int, int, int], str], which pytorch version to use for initializing nn.modules.
+            The format should bea version string (major,minor,micro) or a tuple of integers. Examples of valid
+            version are (1,7,0), '1.7.1'.
+        use_master : bool, If True, then the latest initialization scheme is used ignoring the value of `version`
```

Comment:
it's not clear what this means to me.  My interpretation of how this is written is this will go somewhere and download the latest and greatest initialization scheme independent of the version of pytorch that I'm using.  I doubt you meant that though.

Why can't we just pass in torch.__version__ here to get the latest?

Reply:
Removed `use_master`. It was duplicate functionality.